### ISSUE TITLE
Laravel 5.2 paginate 

### ISSUE THREAD

**Issue by kemal-bay** [elasticquent/Elasticquent#33]
Hello everyone,

I'm new to Laravel and elastic search. Nowadays dealing with paginate function.
Let's say, we're trying to get books with hard cover.

`$books = Book::searchByQuery(array('match' => array('cover' => 'hard')))->paginate(12);`

In the first page, the code looks like it's working. Laravel paginator works in view, there're results that I'm looking for. 

But when I check the results, there's a problem with result count. I asked 12 results per page. As you can see, it returned 10 results which is default value for size parameter, right?

![firstpage](https://cloud.githubusercontent.com/assets/15131705/12994165/28832ac2-d127-11e5-94ec-5b405ca59a6b.png)

The most important problem starts when you try to reach other pages (e.g. /?page=2). Although you can see the pagination links on the view, the collection is empty.

![fourthpage](https://cloud.githubusercontent.com/assets/15131705/12994166/288745c6-d127-11e5-8fc1-0f010a2dc421.png)

As i mentioned earlier, i'm new to this technologies but imho paginate function must set from and size parameters before making a request to elastic search server.

Am I missing something? Could you please enlighten me about the current pagination process?

Thanks,

**Comment by kemal-bay** [comment]
Hello again! I combined two functions, "complexSearch" and "paginate" in my fork.
[https://github.com/kemal-bay/Elasticquent/commit/4d9e02baf42d8d9619ceb5626d8e1eb28287f667](url)

I'd like to make a pull request if it's ok for you.

**Comment by BertvanHoekelen** [comment]
@kemal-bay Please make the pull request. I think it's a nice contribution.

**Comment by timgws** [comment]
@kemal-bay I am with @BertvanHoekelen, however, please make sure that you do not remove functions (I see in this commit that you have removed `paginate` and `getAggregations`). This is to ensure we keep compatibility with existing codebases that may already be using these functions.

**Comment by kemal-bay** [comment]
@timgws Yes, you're totally right. I'll make the changes and will make a pull request as soon as possible. Thanks.

**Comment by kemal-bay** [comment]
I made a pull request and thank you for your supports.

**Comment by nicksmetBagaar** [comment]
+1
